### PR TITLE
Preliminary implementation of Period Chart

### LINKED ISSUE #11063: [Stats Refresh] Days / Weeks / Months / Years : Construct Chart of Views for a Given Period
Stats offers views of a site's performance in various timeframes, including Days, Weeks, Months & Years (DWMY). Theses views call for a Chart highlighting performance of a given site along the several dimensions, including Views.

 The chart should allow the user to accomplish the following:

- Show a bar graph of views with available data.
- Adjust chart scale and dates relevant to data displayed
- Graphically highlight the selected period on the chart.
- Support tapping on another bar to refresh the screen for the newly selected period.

This chart will rely on similar stubbed & actual data, but will more closely resemble the [Selected Post chart](https://github.com/wordpress-mobile/WordPress-iOS/issues/11062) than the [LPS chart](https://github.com/wordpress-mobile/WordPress-iOS/issues/11061).

### PR DESCRIPTION
**Pull request by stevebaranski**
### Description
Fixes #11063, which calls for the introduction of chart styling for period data, with an emphasis on the overlay of Visitors / Views data. 

To test:

- Checkout the branch, confirm that it builds & tests pass.
- For a site, navigate to Stats, and select a given period (i.e., Days, Weeks, Months, Years). 
- Please note that at the moment, the chart is the same regardless of period or dimension (Views, Visitors, Likes, Comments).

Update release notes:

- [X] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.

### Exhibits

11063pr

_iPad - Portrait_
11063pr_ipadportrait

_iPad - Landscape_
11063pr_ipadlandscape

_iPhone SE_
11063pr_iphonese

### CODE REVIEW

**Comment by stevebaranski** [comment]
@ScoutHarris would you be willing to take a look?

@SylvesterWilmott I have included a few screenshots depicting the chart in the proper place. While some of the filter bar wrapping / clipping observed in #11204 remains, hopefully this will give you a sense of things.

In the case of the overlaid bars, I had to adjust the opacity of the highlight so that the Visitors bar was visible. This washes out a bit. I'm open to any suggestions on how I could make this better.

**Comment by frosty** [comment]
> In the case of the overlaid bars, I had to adjust the opacity of the highlight so that the Visitors bar was visible. This washes out a bit. I'm open to any suggestions on how I could make this better.

I wonder if perhaps we need a `secondaryBarHighlightColor`, and have them both fully opaque? I can't see how we'll get the desired colors using translucency. 🤔

**Comment by SylvesterWilmott** [comment]
@planarvoid How was this implemented on Android?

**Comment by stevebaranski** [comment]
> I wonder if perhaps we need a `secondaryBarHighlightColor`, and have them both fully opaque? I can't see how we'll get the desired colors using translucency. 

@frosty thanks for this suggestion! I've observed that a single highlight appears to be higher in the z-order than both bars. I had not considered the notion of two highlights. I will give this prompt attention on Monday. I have updated the milestone associated with the PR to reflect this.

**Comment by frosty** [comment]
@stevebaranski I'm not sure if I'm using the right terminology – the marker is the translucent background that stretches to the top? That one appears to be behind both bars, is that correct? Is that different from the single highlight?

**Comment by stevebaranski** [comment]
@frosty 

> the marker is the translucent background that stretches to the top?

Yes. It is translucent, and extends from the top of the bar to the top of the chart. In the framework, it's a custom view (hence, the rotation issue you observed before). Other uses might include a "bubble", or even a label with the specific data value.

> Is that different from the single highlight?

Yes. The highlight uses the existing bar as a mask. In the case where Views/Visitors are presented together, highlighting is enabled for the View bar, but disabled for Visitors. Unfortunately, however, the highlight occludes the Visitors bar.

@SylvesterWilmott was right to suggest that I reach out to @planarvoid; I've done that via Slack, and will consider his input on Monday.

Depending on what that change entails, I'll see if it makes sense to add commits to this PR, or to consider a change in a subsequent PR. As it stands, the PR can still be reviewed, but the aesthetics may improve.

File: WordPress/Classes/ViewRelated/Stats/Period Stats/Overview/OverviewCell.swift [hunk 1]
```diff
@@ -97,4 +131,26 @@ private extension OverviewCell {
...
+
+    // MARK: Chart support
+
+    func resetChartView() {
+        for subview in chartContainerView.subviews {
+            subview.removeFromSuperview()
+        }
+    }
+
+    func configureChartViewIfNeeded() {
```
Comment (ScoutHarris) [inline WordPress/Classes/ViewRelated/Stats/Period Stats/Overview/OverviewCell.swift]: I'm a little confused by this method name, specifically the `IfNeeded` part. Is the chart ever _not_ needed on the Overview? There isn't a condition within the the method that indicates that scenario.

**Comment by ScoutHarris** [comment]
Hey @stevebaranski . I left just one minor comment/question. I'll wait to give a final review until you decide if any additional changes are coming.

**Comment by planarvoid** [comment]
hey @stevebaranski, the code handling this functionality is in `org.wordpress.android.ui.stats.refresh.lists.sections.viewholders.BarChartViewHolder`. 
What I do is I have 3 data sets. A base data set with `Views` which has `blue_wordpress` color and `orange_active` as highlight color. A second data set (`Visitors`) which has `blue_dark` color and `orange_dark_active` as highlight color (`highlightAlpha` is set to `255` in both cases). The last data set is what I call a Highlight data set in which I manually set the values to max value of the chart (so it fills the whole column. This data set has a default color set to `transparent` so it's invisible by default and its highlight color is `orange_active` with alpha set to `51`. 
Be careful when you add the data sets to the `BarData` because the order matters. I tried loads of variations and in the end I add first the default `Views`, then the `Visitors` and then the `Highlight`. 
The real magic is how I handle highlighting. The default highlight doesn't work with multiple datasets (especially since one is invisible). I'm setting my custom `OnChartValueSelectedListener` which calls a method `highlightColumn` which does this:
```
if (hasOverlappingColumns) {
            val high = Highlight(index.toFloat(), 0, 0)
            val high2 = Highlight(index.toFloat(), 1, 1)
            val high3 = Highlight(index.toFloat(), 2, 2)
            high.dataIndex = index
            high2.dataIndex = index
            high3.dataIndex = index
            highlightValues(arrayOf(high3, high, high2))
        } else {
            val high = Highlight(index.toFloat(), 0, 0)
            val high2 = Highlight(index.toFloat(), 1, 1)
            high.dataIndex = index
            high2.dataIndex = index
            highlightValues(arrayOf(high2, high))
        }
```

As you see, I'm setting the highlight manually to the data sets. Be careful, the order matters! It took me ages to get it right. This works with datasets set to [Views, Visitors, Highlight]. Let me know if this helps!

**Comment by SylvesterWilmott** [comment]
@stevebaranski this all looks so great! Thanks for spending the time to perfect the charting.

In landscape mode and tablet will we increase the number of bars we are showing?

**Comment by stevebaranski** [comment]
Hey @SylvesterWilmott - thanks for this feedback. 

> In landscape mode and tablet will we increase the number of bars we are showing?

At the moment, the stubbed data has 10 values. Once we start to integrate with the "live" data set, I'd like to work with you to optimize the appearance for each device. Potentially, we can control the number of rows that appear, or fine tune the bar width, etc. If you want to keep an eye on the relevant issues, #11067, #11068, #11069 are the ones I have in mind.

File: WordPress/Classes/ViewRelated/Stats/Period Stats/Overview/OverviewCell.swift (hunk 1 above)
Comment (stevebaranski) [reply WordPress/Classes/ViewRelated/Stats/Period Stats/Overview/OverviewCell.swift]: This is valid feedback. I've pushed a commit ([`3779c0a`](https://github.com/wordpress-mobile/WordPress-iOS/pull/11210/commits/3779c0a6d9e499c94047fb9fa85bb9449b2c3073)) that removes the `IfNeeded` suffix.

**Comment by stevebaranski** [comment]
@ScoutHarris I have addressed your feedback with an additional commit. I have also created #11230 to capture the styling changes discussed herein. I will tackle that next, but would like to perform that work in a separate PR.

Do you mind revisiting this review?

**Pull request merged**